### ISSUE TITLE
Running custom model with tensorRT error

### ISSUE THREAD

**Issue by winter2897** [haotian-liu/yolact_edge#126]
Could you help me! @haotian-liu 

# 1. Describe the bug
## 1.1 Running on PC

**Error:**  [IndexError]: too many indices for tensor of dimension 2

```
python eval.py --trained_model=weights/yolact_edge_mobilenetv2_config_for_rtr_thannong_700_476_10000.pth --config=yolact_edge_mobilenetv2_config_for_rtr_thannong_700_config --score_threshold=0.5 --top_k=100 --video_multiframe=2 --trt_batch_size 2 --video=lua720.mp4 --use_fp16_tensorrt
```

![PC_mymodel_trt](https://user-images.githubusercontent.com/22476495/123617239-3dfe3880-d831-11eb-8b9f-90886a08c8e4.png)

I followed your instructions at issue #27. However, the error still occurs. Everything runs only when I `-- disable tensorrt`.

## 1.2 Running on Jetson NX

**Error:** Segmentation fault (core dumped)

```
python eval.py --trained_model=weights/yolact_edge_mobilenetv2_config_for_rtr_thannong_700_476_10000.pth --config=yolact_edge_mobilenetv2_config_for_rtr_thannong_700_config --score_threshold=0.5 --top_k=100 --video_multiframe=2 --trt_batch_size 2 --video=lua720.mp4 --use_fp16_tensorrt
```

![coreDump](https://user-images.githubusercontent.com/22476495/123617763-cbda2380-d831-11eb-8e11-8a2b88fbb0ac.png)

I have followed your instructions in Issue #45, set `cfg.torch2trt_max_calibration_images  =  5` . However, the error still occurs.

![set5](https://user-images.githubusercontent.com/22476495/123622557-8ec46000-d836-11eb-9b75-83db44b5e5b1.png)

# 2. To Reproduce

Create a new custom dataset:

```
RTR_thannong_06_700_dataset = dataset_base.copy({
    'name': 'RTR thannong 700',
    
    'train_images': './data/thannong_06_700/images/',
    'train_info': './data/thannong_06_700/train.json',
    
    'valid_images': './data/thannong_06_700/images/',
    'valid_info': './data/thannong_06_700/test.json',

    'has_gt': True,

    'class_names': ('object', 'field', 'construction', 'person', 'animal'),
    'label_map': {1:1, 2:2, 3:3, 4:4, 5:5}
})
```
Config dataset:
```
yolact_edge_mobilenetv2_config_for_rtr_thannong_700_config = yolact_edge_config.copy({
    'name': 'yolact_edge_mobilenetv2_config_for_rtr_thannong_700',

    # Dataset stuff
    'dataset': RTR_thannong_06_700_dataset,
    'num_classes': len(RTR_thannong_06_700_dataset.class_names) + 1,

    'lr_steps': (2800, 6000, 7000, 7500),
    'max_iter': 8000,
    'backbone': mobilenetv2_backbone
})
```

Pretrain model : `mobilenet_v2-b0353104.pth`

Traning command:

```
python train.py --batch_size=8 --num_workers=4  --config=yolact_edge_mobilenetv2_config_for_rtr_thannong_700_config --dataset=RTR_thannong_06_700_dataset --resume=weights/mobilenet_v2-b0353104.pth --log=./logs
```

Full log training (no error): 
[log.txt](https://github.com/haotian-liu/yolact_edge/files/6725362/log.txt)

# 3. Expected behavior

@haotian-liu When I use your model [yolact_edge_mobilenetv2_54_800000.pth](https://drive.google.com/file/d/1L4N4VcykqE-D5JUgWW9zBd6WKmZPBAZQ/view). I don't get any `TensorRT` or `Core dumped`  related errors. I wonder what setup the difference comes from in the training process?

## 3.1 Running on PC: 

```
python eval.py --trained_model=weights/coco/yolact_edge_mobilenetv2_54_800000.pth --score_threshold=0.5 --top_k=100 --video_multiframe=2 --trt_batch_size 2 --video=person720.mp4 --use_fp16_tensorrt
```
![PC_origin_trt](https://user-images.githubusercontent.com/22476495/123620552-908d2400-d834-11eb-80ff-b0fd59c7014c.png)

## 3.2 Running on NX

```
python eval.py --trained_model=weights/coco/yolact_edge_mobilenetv2_54_800000.pth --score_threshold=0.5 --top_k=100 --video_multiframe=2 --trt_batch_size 2 --video=person720.mp4 --use_fp16_tensorrt
```
![origin_model](https://user-images.githubusercontent.com/22476495/123620646-a8fd3e80-d834-11eb-85c8-ddd0d00a6863.png)

# 4. Additional context

When I don't use `tensorRT`  there is no error on both `PC` and `Jetson NX`. However, I want to use `TensorRT` to increase FPS on `Jetson NX`.

||PC|NX|
|-|-|-|
|FPS|46|6|
```
python eval.py --trained_model=weights/yolact_edge_mobilenetv2_config_for_rtr_thannong_700_476_10000.pth --config=yolact_edge_mobilenetv2_config_for_rtr_thannong_700_config --score_threshold=0.5 --top_k=100 --video_multiframe=2 --trt_batch_size 2 --video=lua720.mp4 --disable_tensorrt
```

![PC_mymodel_no_trt](https://user-images.githubusercontent.com/22476495/123621567-8fa8c200-d835-11eb-98af-4a334a7890f5.png)

# 5. Environment

```
Pytorch: 1.6.0
torch2trt: 0.1.0
TensorRT: 7.1.3
cuDNN: 8.0
CUDA: 10.2
OpenCV: 4.1.1
```

**Comment by haotian-liu** [comment]
Hi, have you tried safe mode as described [here](https://github.com/haotian-liu/yolact_edge#handling-inference-error-when-using-tensorrt)?

**Comment by winter2897** [comment]
@haotian-liu I tried safe mode and it still doesn't work.

**Comment by winter2897** [comment]
@haotian-liu Thanks for the suggestion, I tried it and it worked fine on Jetson NX. Also can you explain the difference between the two models as mentioned above?

![good](https://user-images.githubusercontent.com/22476495/123782182-3d33d800-d8ff-11eb-8752-5cce80f32505.png)

**Comment by haotian-liu** [comment]
@winter2897 When we need to filter out the detections, we typically do masks to select them: having a boolean tensor `mask` and select them with `t[mask]`. It fails sometimes with TensorRT with some models, I will then use some PyTorch functions to do the same thing (which I found to be able to alleviate such issue. It's some emperical workaround to resolve the issue (and all the models are exactly the same). Hope this explains and clears your doubts, thanks!

**Comment by winter2897** [comment]
@haotian-liu thank you!
